MoreLinq.Test/BatchTest.cs

```diff
@@ -133,6 +133,16 @@ public void BatchEmptySource(SourceKind kind)
             var batches = Enumerable.Empty<int>().ToSourceKind(kind).Batch(100);
             Assert.That(batches, Is.Empty);
         }
+
+        [Test]
+        public void BatchUsesCollectionCountAtIterationTime()
+        {
+            var stack = new List<int>(Enumerable.Range(1, 3));
+            var result = stack.Batch(3);
+            stack.Add(4);
+            result.AssertCount(2).Consume();
+            Assert.Pass();
```

Comment:
[Follow-up](https://github.com/morelinq/MoreLINQ/pull/965#discussion_r1102691206) suggestion:

```suggestion
        [TestCase(SourceKind.BreakingList)]
        [TestCase(SourceKind.BreakingReadOnlyList)]
        [TestCase(SourceKind.BreakingCollection)]
        public void BatchCollectionSmallerThanSizeUsesCollectionCountAtIterationTime(SourceKind kind)
        {
            var list = new List<int> { 1, 2 };
            var result = list.ToSourceKind(kind).Batch(3);
            list.Add(3);
            result.AssertSequenceEqual(new[] { 1, 2, 3 });
        }
```

Reply:
Rejected per [comment](https://github.com/morelinq/MoreLINQ/pull/965#discussion_r1103908079)